### PR TITLE
Si 4348 apm for module discovery

### PR DESCRIPTION
**Pull request by kivkiv12345**
This branch currently only changes known_hosts to become a linked list, which I would prefer when working with PyCSH.
So I suggest a merge now, so CSH and PyCSH receive the updated implementation simultaneously.

### CODE REVIEW

File: src/known_hosts.h
Comment (jeanbaptistelab) [inline src/known_hosts.h:1]: If we're going to #error when PARAM_HAVE_SYS_QUEUE is not defined, we might as well remove the option to build without it in the meson setup and maybe sweep the CSH code to remove the corresponding "#ifdef PARAM_HAVE_SYS_QUEUE" statements, no ?

Comment (kivkiv12345) [reply src/known_hosts.h:1]: That depends on whether we want to support known_hosts being optional.
I recognize that the required checks can be error-prone, and that they will bloat the code somewhat.
Still, I can't help but to like the modularity it brings.
In the end, I have no strong feelings one way or the other. I understand you are cautious of modular conditional compilation @jeanbaptistelab ?

Comment (jeanbaptistelab) [reply src/known_hosts.h:1]: It's not so much the error-prone nature that tickles me, it's more the (mis)use of a configuration item ("PARAM_HAVE_SYS_QUEUE") from a dependency to en/dis-able a feature ("known_host") in CSH... 

Looking at the code, it seems to only be using features from "libbsd", not anything from "libparam with PARAM_HAVE_SYS_QUEUE enabled" so if you want to make the "known_host" a toggle-able feature, I would suggest instead adding a new meson item (and a corresponding macro like "CSH_ENABLE_HOSTNAMES" or something) which enables it AND requires the presence of "libbsd-overlay", much in the same way as in libparam's meson setup.

Anyways, I don't have a strong opinion on that, I was just pointing out that in its current installment, the code simply cannot build without "PARAM_HAVE_SYS_QUEUE" (I tried it), effectively changing PARAM_HAVE_SYS_QUEUE from an option to a requirement, and ipso facto making "known_host" not a option, but rather a side-effect of building CSH with "param:list_dynamic=true" or "param:list_pool=true"...

Just my 2 cents :)

**Comment by kivkiv12345** [comment]
I have removed the checks for PARAM_HAVE_SYS_QUEUE.
I felt conflicted about removing the #error, but decided that it may end up lying in the future.

**Pull request merged**